content/docs/1.6.2/high-availability/recover-volume.md

```diff
@@ -1,13 +1,40 @@
 ---
-  title: Recover Volume after Unexpected Detachment
+  title: Volume Recovery
   weight: 1
 ---
 
-When an unexpected detachment happens, which can happen during a [Kubernetes upgrade](https://github.com/longhorn/longhorn/issues/703), a [Docker reboot](https://github.com/longhorn/longhorn/issues/686), or a network disconnection,
-Longhorn automatically deletes the workload pod if the pod is managed by a controller (e.g. deployment, statefulset, daemonset, etc...).
-By deleting the pod, its controller restarts the pod and Kubernetes handles volume reattachment and remount.
+Longhorn provides two mechanisms for maintaining volume functionality in a variety of situations.
 
-If you don't want Longhorn to automatically delete the workload pod, you can set it in [the setting `Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) in Longhorn UI.
+## Automatic Workload Pod Deletion
 
-For the pods that don't have a controller, Longhorn doesn't delete them because if Longhorn does, no one will restart them.
-To recover unexpectedly detached volumes, you would have to manually delete and recreate the pods that don't have a controller.
+This recovery mechanism is enabled by the setting [*Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly*](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly).
+
+When one of the following situations occurs, Longhorn automatically attempts to delete workload pods that are managed by a controller (for example, Deployment, StatefulSet, or DaemonSet). After deletion, the controller restarts the workload pod and Kubernetes handles volume reattachment and remounting.
+
+1. A volume was unexpectedly detached, possibly because of a [Kubernetes upgrade](https://github.com/longhorn/longhorn/issues/703), [container runtime reboot](https://github.com/longhorn/longhorn/issues/686), network connectivity issue, or volume engine crash.
+2. A volume was automatically salvaged after all replicas became faulty, possibly because of a network connectivity issue. Longhorn attempts to identify the usable replicas and uses them for the volume.
+3. An error occurred on a Share Manager pod that uses an RWX volume.
```

Comment:
note: Item 3 should not be the case if enabling [RWX HA](https://longhorn.io/docs/1.7.0/high-availability/rwx-volume-fast-failover/) which is supported from v1.7.0.  

Let's make sure v1.7 docs have updates.